### DISCUSSION TITLE
Ser franco consigo mesmo e saber mensurar suas Skills

### DISCUSSION THREAD

**Discussion by igorthurow** [frontendbr/forum#1208] [Discussão]
Boa tarde!

Para fazer sentido: Sou Júnior!

Pensando em atualizar meu CV, me surgiu uma dúvida: Como mensurar minhas próprias Skills? 

Posso dizer que sei tal tecnologia por x tempo de experiência? Ou por tal curso?

Penso em atingir as expectativas dos entrevistadores. 
É claro que um desafio é ótimo, mas devo conseguir progredir neste desafio, certo?! Ao ver vagas, fico em dúvida se de fato sei tal tecnologia a nível de preencher uma vaga de forma satisfatória, ou se não devo me preocupar tanto com isso e arriscar me quebrar e aprender algo novo.

Um exemplo é o que eu vivo hoje: Há 5 meses codo ReactJS com Type em um novo projeto. Eu sou júnior, e dentro do que é proposto eu atinjo as expectativas. Mas fico com esta pulga atrás da orelha para quando eu quiser desbravar novos desafios.

Valeu!

**Comment by CassioGangrel** [comment]
Kra tenho as msm duvidas,  ficarei de olho nas respostas. Eu to com 29 anos, começando a programar agora com JS, quando criança eu fazia muitos bots para mmo adorava mas para mim era brincadeira, nao entendia na época e acabei estudando e trabalhando com eletronica. Hj resolvi resgata a alegria de programar de quando era criança. Mas tenho esta dificuldade em saber quando estarei pronto para migrar de emprego. XD

**Comment by cauequeiroz** [comment]
@igorthurow, bem interessante o seu questionamento. Na minha opinião isso não é algo que você deva se preocupar muito, é tudo uma questão de contexto e, claro, muita conversa.

O simples já serve bem: **coloca no seu currículo as tecnologias que você se sente confortável para trabalhar.** 

> Mas Caue, eu comecei a estudar React agora, não tenho conhecimento para pegar um projeto avançado de ponta a ponta e programar a interface do software que vai mandar o homem para marte, o que eu faço?

É ai que entra o contexto, meu caro. Quando uma empresa abre uma vaga para junior, ela espera que o candidato tenha um conhecimento básico das tecnologias que ele diz saber. Da mesma forma, quando a empresa abre uma vaga para senior, ela espera que o candidato tenha um conhecimento bastante profundo de tais tecnologias.

Então, se você esta se candidatando para uma vaga Junior, fica tranquilo. Você não precisa ser o chuck norris do React e o seu conhecimento um pouco mais básico já servirá.

Não vejo necessidade de você _mensurar_ de alguma forma o quanto você sabe de algo. Na verdade eu acho bem ridiculo até quando a galera resolve fazer aquelas barrinhas com porcentagem tipo _"90% 
 html, 85% css, 57% javascript"_. O que diabos é alguém saber 57% de javascript? haha.

Aprendeu uma tecnologia nova como o React? Legal. Faz um projeto pessoal, coloca no seu github, coloca a tecnologia no seu currículo, **e é isso**. A empresa vai entender que você sabe e vai poder, através do seu projeto no github, avaliar de forma mais precisa e decidir se o seu conhecimento já serve para vaga.

**Comment by wmartins** [comment]
@igorthurow, vou responder usando bastante React como exemplo, até pra trazer um pouco mais próximo da tua realidade atual, mas isso se aplica para qualquer tecnologia/função.

> **Penso em atingir as expectativas dos entrevistadores.**

Acho que aqui vale muito entender quais são essas expectativas e o que tu vai precisar fazer na prática. Por exemplo, tu pode ser muito bom em React e já trabalhar há anos com isso, porém, o projeto pede para saber como fazer componentes que renderizem gráficos. O quão confortável tu fica com isso? Se não, tu tem disposição/tempo/vontade de aprender?

Acho que é um pouco disso que o entrevistador vai querer saber de ti. Cada funcionário novo é um investimento para a empresa (seja em tempo, dinheiro e enfim), assim, a questão é ver o quanto tu pode agregar e se isso é uma coisa que vale (tanto pra ti quanto pra empresa).

> **Posso dizer que sei tal tecnologia por x tempo de experiência? Ou por tal curso?**

Isso é bem relativo. Tem pessoas que estudam uma tecnologia em uma semana e já conseguem ser bem produtivas, já algumas outras pessoas ficam um tempão com uma mesma tecnologia e têm dificuldades em entender algumas coisas. Enfim, é algo muito pessoal. Provavelmente fazer um curso vai te ajudar a passar por coisas relevantes de uma tecnologia, mas não significa que tu sabe ela.

Pegando React por exemplo, eu, particularmente, teria algumas coisas que eu classificaria por nível de dificuldade (essas são opiniões pessoais minhas):

- JSX (iniciante)
- `prop` e `state` (iniciante)
- ...
- High order components (intermediário)
- Portals (intermediário)
- ...
- Server rendering (estratégias, *trade offs*) (avançado)
- Algoritmo de reconciliação (avançado)
- Criação de custom renderers (avançado)
- ...

Com isso, alguém que sabe muito de React deveria saber tudo do básico, boa parte das coisas do intermediário e algumas do avançado (pelo menos ter ouvido falar ou ter uma noção de como funciona). Mas isso é não é regra.

> **Para fazer sentido: Sou Júnior!**

Não te apega muito nisso, por exemplo, têm uma galera que é Senior, mas que quando se trata de uma tecnologia específica, ainda está engatinhando. A maior diferença é que uma pessoa Senior já vai saber os "caminhos" de onde estudar, de separar o que é relevante e o que não é e, por ter mais experiência, vai ter familiaridade com alguns conceitos que a tecnologia tem.

Acho que a questão de senioridade não tá tão ligado com tecnologias, tá mais com pessoas, processos, comportamento e etc. Acho que é um assunto pra outra hora hahaha.

> **Mas fico com esta pulga atrás da orelha para quando eu quiser desbravar novos desafios.**

Acho que uma forma pra te ajudar é justamente entender a tecnologia como um todo. Por exemplo, das coisas listadas em Main Concepts nos [docs do React](https://reactjs.org/docs), o que dessas coisas tu nunca usou ou nunca ouviu falar? Essas coisas te tomariam muito tempo pra aprender? Se tu manja tudo, vê as coisas no Advanced Guides e repete as perguntas.

Enfim, se quiser trocar mais uma ideia sobre é só avisar. :)
